plugins/linux-v4l2/v4l2-output.c

```diff
@@ -125,19 +138,57 @@ static bool try_connect(void *data, const char *device)
 
 	vcam->device = open(device, O_RDWR);
 
-	if (vcam->device < 0)
+	if (vcam->device == -1) {
+		blog(LOG_ERROR, "Failed to open device '%s' (%d)", device,
+		     errno);
 		return false;
+	}
+
+	if (ioctl(vcam->device, VIDIOC_QUERYCAP, &capability) == -1) {
+		blog(LOG_WARNING,
```

Comment:
Would it be possible to store an error message in string here and then add an output call after the `fail_close_device` label instead? That label seems to serve as the bail-out point for all errors after this point.